### ISSUE TITLE
Build failed on Linux, possible cause

### ISSUE THREAD

**Issue by TheForgetTime** [xamarin/xamarin-android#3910]
## 1. The git submodule repository is not properly initialized
Console output project does not exist when executing make prepare.

Some git submodules are not checked out.
>What does not exist when I build is (`nrefactory`)

The program continues to run after manual checkout.

Then there was a second error.

## 2. Although the nuget package has been installed, it does not work.

A part of the nuget package has no effect after being referenced.
>My reference to the build is not normal :
>  `Mono.Cecil` in `Mono.Debugger.soft` and `Mono.Debugging.soft`
>  `Microsoft.CodeAnalysis(.*)` in `Mono.Debugging.soft`
>  And other

The console output did not find the namespace when running the make prepare command.

This was successfully built after I used the IDE to manually reference the assembly in the nuget package.

### Environment
HOST OS:
>Archlinux x86_64 rolling

HOST MONO VERSION: 
>Mono JIT compiler version 6.0.0 (makepkg/6256b82d62f Fri 20 Sep 2019 11:18:59 PM -03)

HOST MSBUILD VERSION: 
>Microsoft (R) build engine version 16.0.0.0 for Mono

**Comment by TheForgetTime** [comment]
And passed all the tests in the Debug version.

**Comment by pjcollins** [comment]
@TheForgetTime would you be able to provide the binary logs or xaprepare logs for the build failures you are encountering? Alternatively, if you've managed to address these issues locally feel free to send a pull request 😄

**Comment by grendello** [comment]
@TheForgetTime Arch support is most likely incomplete in the current `master`, a PR would really be welcome :) Please take a look at [this file](https://github.com/xamarin/xamarin-android/blob/fbe4b44c7b40dcaacdecaed834c631b15490cc52/build-tools/xaprepare/xaprepare/ConfigAndData/Dependencies/Linux.Arch.cs) to see if you can spot anything that's missing

Submodule initialization failure is often a sign of some corrupted data in your `.git` directory, please try the following in your copy of the repository to see if it works (with no local changes):

```bash
$ git pull --rebase
$ git clean -xdff
$ git submodule foreach --recursive git clean -xdff
$ git submodule foreach --recursive git reset --hard HEAD
$ git submodule sync
$ git submodule update --recursive --init --force
```

**Comment by TheForgetTime** [comment]
@pjcollins I can't copy any files from my company's computer due to various factors.
So I decided to try it on my PC, this process may take a long time due to network reasons.
I can provide these files after this.

**Comment by TheForgetTime** [comment]
@grendello All packages listed in the file have been installed correctly.
The main reason I want to cause file corruption in `.git` is that my country's firewall blocked some files.

For reference to querying and installing missing packages on Archlinux, I referenced the [xamarin-android-git](https://aur.archlinux.org/packages/xamarin-android-git/) package in Aur

**Comment by TheForgetTime** [comment]
First apply the xaprepare-arch.patch file in aur/xamarin-android-git
First execution `make prepare-external-git-dependencies`
>
>“/home/auser/xamarin-android/build-tools/xaprepare/xaprepare.sln”(默认目标) (1) ->
“/home/auser/xamarin-android/build-tools/xaprepare/xaprepare/xaprepare.csproj”(默认目标) (2) ->
(CoreCompile 目标) -> 
  Main.cs(7,12): error CS0234: The type or namespace name 'Options' does not exist in the namespace 'Mono' (are you missing an assembly reference?) [/home/auser/xamarin-android/build-tools/xaprepare/xaprepare/xaprepare.csproj]
  Steps/Step_Android_SDK_NDK.cs(7,7): error CS0246: The type or namespace name 'Kajabity' could not be found (are you missing a using directive or an assembly reference?) [/home/auser/xamarin-android/build-tools/xaprepare/xaprepare/xaprepare.csproj]
>
>    0 warning
>    2 Error

### The changes I made
Remove the nuget package reference and manually add the locally cached assembly in net40

**Comment by TheForgetTime** [comment]
Second execution `make prepare-external-git-dependencies`
Monodroid (https://github.com/xamarin/monodroid/) project cannot be cloned
Skip this step and execute make prepare directly

**Comment by TheForgetTime** [comment]
Sorry, the build was interrupted because Corretto JDK could not be downloaded.
The next reply and log will continue after I download to Corretto JDK through various channels.

In fact, the remaining errors are that the nuget package cannot be applied.
I also only performed the following operations.
Remove the nuget package reference and manually add the locally cached assembly in net40(The one in the corresponding package in the nuget cache instead of the global environment)

**Comment by grendello** [comment]
@TheForgetTime the `prepare-external-git` target should not be run, it's for internal builds only.

Regarding the other issues you reported so far, I don't see anything we can do on our side as long as your problems revolve around restoring nugets :( We can't remove the nuget references as they are needed for the build and Xamarin.Android to work :(

**Comment by jonathanpeppers** [comment]
@TheForgetTime let us know if you need us to look into something else here, thanks!

**Comment by TheForgetTime** [comment]
All issues I encountered have been resolved on 10.3.0.33
Build on 10.3.0.33 has been successful, although I only compiled Android 10, but I think other Android versions should not be a problem

In issues #4432 I provided a download link for a Chinese Baidu network hard drive

thank you for your attention to this issue @jonathanpeppers
